### ISSUE TITLE
Accessibility conformance commitment level

### ISSUE THREAD

**Issue by mmikitka** [zurb/foundation-sites#10348]
I am working on some sites for a public institution which must satisfy WCAG 2.0 AA, and I am wondering whether the Foundation community would consider striving for a minimum WCAG 2.0 conformance level for all components, and publish WCAG 2.0 conformance test results based on the [WCAG Evaluation Methodology](https://www.w3.org/TR/WCAG-EM/).

It seems as though accessibility-related legislation is making greater demands on public and private institutions, and striving for conformance now may pay-off in the future when accessibility conformance demand is greater. And, aside from legislation, attention to accessibility concerns can improve the overall usability of Foundation components from a UI perspective and a data consumer perspective (data enrichment, in the form of semantics, is co-evolving with the accessibility movement).

I only came across one UI component framework with a scope comparable to Foundation and Bootstrap, and a firm commitment to WCAG 2.0 conformance: [WET-BOEW](https://github.com/wet-boew/wet-boew) is committed to WCAG 2.0 AA. However, I believe that WET-BOEW is lacking in the following areas relative to Foundation (Note: I prefer Foundation over Bootstrap):

# Benefits of Foundation over WET-BOEW
* Web user experience
  * Greater diversity of use-cases (Zurb is a design company)
* Code quality
  * Separation of concerns
  * Modularity of components
  * Automated test coverage (WET-BOEW has little to none)
* Extensibility
  * Customization of behaviour
  * Customization of presentation
  * Compatibility with JS frameworks
* Documentation (Foundation's documentation is way better)
* Supportability (Larger community, very responsive)
* Tool integration (Larger app ecosystem)

However, the driving force towards WET-BOEW is a firm commitment to WCAG 2.0 AA.

I have been using Foundation for five years, and I want to continue to use it, but the cost-benefit of a framework like WET-BOEW is lower in regions where accessibility demands are greater, and in my current case, is pushing us away from Foundation.

Thanks for considering.

**Comment by kball** [comment]
Hi @mmikitka,
Foundation absolutely is dedicated to accessibility, was rebuilt in 6.0 to allow automatic aria manipulation for all JS components, and in fact in 6.3 change default color pallets to meet WCAG 2.0 AA standards (see https://github.com/zurb/foundation-sites/pull/9319 and https://github.com/zurb/foundation-sites/issues/7387)

If we're falling short on any areas of compliance, we certainly want to hear about them. Can you highlight the problem areas you're running into?

CC @andycochran @Owlbertz

**Comment by andycochran** [comment]
Yeah, I'd love to work on any specific areas where Foundation isn't meeting AA requirements. Please create issues for anything you notice.

**Comment by mmikitka** [comment]
I appreciate the quick response and enthusiasm.

It seems as though the current approach to accessibility conformance testing is ad-hoc: if you stumble across something, then log it. However, I am proposing a more controlled approach, inspired by the (WCAG Evaluation Methodology)[https://www.w3.org/TR/WCAG-EM], to (a) discover non-conformance at a intra-component and inter-component level, (b) regression tests, cognizant of the need for manual and automated tests (some Sufficient Techniques require manual intervention), and (c) a reporting process, ideally by release (e.g., obtain an accessibility conformance report of all components in the 6.3 release)

I can help with the design of such a process, and possibly the implementation.

.

**Comment by brettsmason** [comment]
@mmikitka We would definately like to improve things with here with your help.
How do you think is best we progress this forward?

**Comment by mmikitka** [comment]
@brettsmason Might you be able to comment specifically on the high-level plan that I suggested? Once a high-level plan is clarified *and* there is buy-in from all levels of the organization, the details will follow.

Regards

**Comment by brettsmason** [comment]
@mmikitka Sorry I just wanted to get get back to you as there had been a few weeks without a reply to you 😄 This isn't really my area of expertise - @kball @Owlbertz can either of you help move this forward?

**Comment by Owlbertz** [comment]
Hi @mmikitka,
I agree with you that the need for accessible frontend frameworks is raising and that Foundation should aim for being an accessible solution when creating websites with it. Of course the goal here would be a compliance with an official standard. I believe WCAG 2.0 would be a good choice here as it is (afaik) the most common one.

While I feel like an official commitment can only be made by Zurb, I guess @kball would agree that we are currently already aiming for improving accessibility of Foundation (at least on the long run as there are not many contributors taking care of this topic) and having a commitment (and later on maybe even some kind of certification) would be an improvement for this goal.

The methology you purposed makes sense for me. Of course having an initial, complete list of all issues we have with accessibility will be the hardest part. Once this is done, I think a lot of stuff could be included in the automated unit tests (which is already the case for a lot of things like focus movement or usage of ARIA-attributes). Having manual tests on a regular base would be a good thing to add to the "pre-release" task list to catch any other problems that evolved during the development.

**Comment by DanielRuf** [comment]
How do we plan to improve a11y?

We can at least test with axe, pa11y/koa11y, sonarwhal, Chrome Dev Tools and so on.

**Comment by andycochran** [comment]
I'm going to close this, as it's not really a bug. However, we should definitely be strict about WCAG 2.0 AA standards in v7.
